src/governance-vote.cpp

```diff
@@ -163,44 +70,16 @@ vote_outcome_enum_t CGovernanceVoting::ConvertVoteOutcome(const std::string& str
 
 vote_signal_enum_t CGovernanceVoting::ConvertVoteSignal(const std::string& strVoteSignal)
 {
-    vote_signal_enum_t eSignal = VOTE_SIGNAL_NONE;
-    if(strVoteSignal == "funding") {
-        eSignal = VOTE_SIGNAL_FUNDING;
-    }
-    else if(strVoteSignal == "valid") {
-        eSignal = VOTE_SIGNAL_VALID;
-    }
-    if(strVoteSignal == "delete") {
-        eSignal = VOTE_SIGNAL_DELETE;
-    }
-    if(strVoteSignal == "endorsed") {
-        eSignal = VOTE_SIGNAL_ENDORSED;
-    }
-
-    if(eSignal != VOTE_SIGNAL_NONE)  {
-        return eSignal;
-    }
-
-    // ID FIVE THROUGH CUSTOM_START ARE TO BE USED BY GOVERNANCE ENGINE / TRIGGER SYSTEM
-
-    // convert custom sentinel outcomes to integer and store
-    try {
-        int i = boost::lexical_cast<int>(strVoteSignal);
-        if(i < VOTE_SIGNAL_CUSTOM1 || i > VOTE_SIGNAL_CUSTOM20) {
-            eSignal = VOTE_SIGNAL_NONE;
-        }
-        else  {
-            eSignal = vote_signal_enum_t(i);
-        }
-    }
-    catch(std::exception const & e)
-    {
-        std::ostringstream ostr;
-        ostr << "CGovernanceVote::ConvertVoteSignal: error : " << e.what() << std::endl;
-        LogPrintf(ostr.str().c_str());
-    }
-
-    return eSignal;
+    std::map <std::string, vote_signal_enum_t> mapStrVoteSignals = {
+        {"funding",  VOTE_SIGNAL_FUNDING},
+        {"valid",    VOTE_SIGNAL_VALID},
+        {"delete",   VOTE_SIGNAL_DELETE},
+        {"endorsed", VOTE_SIGNAL_ENDORSED}
+    };
+
+    return mapStrVoteSignals.count(strVoteSignal)
```

Comment:
I would prefer smth like
```
const auto it = mapStrVoteSignals.find(strVoteSignal);
if (it == mapStrVoteSignals.end()) {
    LogPrintf("CGovernanceVoting::%s -- ERROR: Unknown signal %s\n", __func__, strVoteSignal);
    return VOTE_SIGNAL_NONE;
}
return it->second;
```

Reply:
I agree. After reading more about `find` vs `at` vs `operator[]` last night, this does seem to be the more conventional way to do lookups.